lib/reporters/ApiaryReporter.js

```diff
@@ -186,6 +187,8 @@ ApiaryReporter.prototype.configureEmitter = function configureEmitter(emitter) {
       if (error) { return callback(error); }
       const reportUrl = this.reportUrl || `https://app.apiary.io/${this.configuration.apiSuite}/tests/run/${this.remoteId}`;
       reporterOutputLogger.complete(`See results in Apiary at: ${reportUrl}`);
+      // Open test report on the Apiary webpage
+      openBrowser(reportUrl);
```

Comment:
Hmmm:

- What happens if this runs in the CI or on a server? Will it still attempt to open the page?
- What if we're in a minimal Docker image and for some reason `xdg-open` doesn't exist? Will it fail Dredd testing?

I think we shouldn't have this here implicitly, but there should be a way user can opt-in to (or at least opt-out from) the behavior. One approach would be to add an option, but it would have to be an option available only with `--reporter=apiary`, which might not be straightforward with the current configuration parsing.

Another approach would be to have `--reporter=apiary-open`, which would be the same reporter, but with opening of the page.

(Another option would be auto-detection (e.g. by checking `process.TTY` or something), but I wouldn't rely on that fully. Auto-detection is nice as a default value, but not as something user cannot override. If we provide the default value by detection, then we need to provide some `open` and `no-open` options to allow user to override the detection.)

Reply:
We can add an extra flag that would be responsible for test report to be open in a browser. This may affect not only Apiary reporter, but also HTMLReporter, and potentially other things that may store test run results in any format on the drive.

I vote for a new flag for two reasons:

1. Prevent complication of the existing `reporter` flag. Effectively an opt-in behavior of opening a test report in a browser is not a new reporter, so it's not semantically valid to supply it as the value of the `reporter` flag.
1. New things are easier to add/deprecate.

What do you think?